build/util/update_group_policy.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import os
+import shutil
+import hashlib
+
+from brave_chromium_utils import wspath
+
+# Chromium stores all group policy definitions under
+# `src/components/policy/resources/templates/policy_definitions/`
+#
+# The name of the file (minus the extension; ex: TorDisable.yaml => TorDisable)
+# corresponds to an auto-generated entry in:
+# `src/out/<build_type_here>/gen/chrome/app/policy/policy_templates.json
+#
+# That auto-generated value (ex: `policy::key::kTorDisabled`) is referenced
+# when we map to a preference in our policy map:
+# `src/brave/browser/policy/brave_simple_policy_map.h`
+#
+# When the code below is ran (trigger via hook; see src/brave/DEPS) this will
+# copy the group policy files from:
+# `src/brave/components/policy/resources/templates/policy_definitions/`
+# to their expected place in Chromium:
+# `src/components/policy/resources/templates/policy_definitions/`
+#
+
+
+def update_policy_files():
+    policy_dir = wspath(
+        "components/policy/resources/templates/policy_definitions/")
+    with os.scandir(policy_dir) as entries:
+        for entry in entries:
+            if not entry.is_dir():
+                continue
+            src_dir = entry.path
+            src_dir_name = entry.name
+            dst_dir = wspath(
+                f"//components/policy/resources/templates/policy_definitions/{src_dir_name}"  # pylint: disable=line-too-long
+            )
+            shutil.copytree(src_dir,
+                            dst_dir,
+                            dirs_exist_ok=True,
+                            copy_function=copy_only_if_modified)
+
+
+def copy_only_if_modified(src, dst):
+    """Copy file if it doesn't exist or if its hash is different."""
+
+    def file_hash(file_path):
+        with open(file_path, "rb") as f:
+            return hashlib.file_digest(f, "sha256").digest()
+
+    if not os.path.exists(dst) or file_hash(src) != file_hash(dst):
+        shutil.copy2(src, dst)
```

Comment:
please also check that REMOVING a file from the the brave directory won't trigger a build error, because in the current approach we only copy/replace files, but never remove them from the upstream directory.

Reply:
Definitely no problem here - I did test that. Basically, it adds a constant into `//out/<build_type_here>/gen/components/policy/policy_constants.cc` which would then just never get used